### PR TITLE
Add gateway version to zbctl

### LINKED ISSUE #3701: Get gateway version in zbctl
Related to #2981 

Possibly include the gateway version in the existing version command

### LINKED ISSUE #3702: Document GatewayVersion RPC request
**Description**

Update the gRPC documentation to reflect the addition of the GatewayVersion RPC

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

* adds the gateway version to the _status_ command in zbctl
* documents the GatewayVersion RPC
* adds cmd/zbctl/internal/* to the ignored folders in gocompat

## Related issues

closes #3701 
closes #3702 

## Pull Request Checklist

- [ ] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [ ] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [ ] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Review by npepinpe — COMMENTED** [review]
So every time I type `zbctl version` it tries to request something? That's kind of surprising. Could we rather add the gateway version as part of the status command? Or if this is a strict requirement, at least we could cache it based on the URL for some TTL?

**Comment by menski** [comment]
Let's add it to the status command output. We can then in a next step also add the broker versions to our membership Infos and print them in the status command.

**Comment by MiguelPires** [comment]
I moved the gateway version to the status command. The test for this is a bit more complicated than before and uses a complicated regular expression which is not great but let me know what you think

File: clients/go/cmd/zbctl/internal/commands/status.go [hunk 1]
```diff
@@ -39,22 +39,36 @@ var statusCmd = &cobra.Command{
...
+		}
+
+		ctx, cancel = context.WithTimeout(context.Background(), defaultTimeout)
+		defer cancel()
+
+		gatewayVersion := "unavailable"
+		verResp, err := client.NewGatewayVersionCommand().Send(ctx)
 		if err != nil {
 			return err
+		} else if verResp != nil && verResp.Version != "" {
```
Comment (npepinpe) [inline clients/go/cmd/zbctl/internal/commands/status.go]: Is it possible for `verResp` to be nil and also `err` to be nil?

File: clients/go/cmd/zbctl/main_test.go [hunk 2]
```diff
@@ -34,6 +34,12 @@ import (
 
 var zbctl string
 
+const (
+	// NOTE: taken from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+	semVer = `(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*` +
```
Comment (npepinpe) [inline clients/go/cmd/zbctl/main_test.go]: I see what you meant by your comment 😂

**Review by npepinpe — APPROVED** [review]
Looks good, could be optimized to have both requests in parallel if it's simple enough but that's also fine to do later

**Comment by MiguelPires** [comment]
Thanks for the review. This feature might end up being refactored so I won't change the parallel request stuff now because it might be gone soon anyway. If no refactor is made, I'll change it

**Comment by MiguelPires** [comment]
bors retry

**Pull request merged**